### ISSUE TITLE
[Crashlytics] Provide linux binaries for `uploads-symbols`

### ISSUE THREAD

**Issue by aried3r** [firebase/firebase-ios-sdk#6007]
Hey!

We're using fastlane for our apps and a lot of its actions can be used on Linux since there are no Mac or iOS specific packages necessary. It also means that we save quite some billable hours since we use a hosted CI solution that bills 10x the amount for every macOS minute spent (GitHub Actions).

Since the `uploads-symbols` binary seems to be closed source (or I didn't find the source), we were wondering if it was at all possible to have an ELF binary or Shell/Python/Ruby/... script of `upload-symbols` to upload dSYMs?

And if not, is there an API endpoint we could manually upload the files to?

**Comment by samedson** [comment]
Hey @aried3r thanks for reaching out. Unfortunately we won't be able to provide either a Linux-based upload-symbols binary or a direct endpoint in the near future. The reason is that upload-symbols relies on critical macOS libraries to convert the dSYM into a lighter weight format, and we don't have a good way of doing it in our backend (if we were to provide a direct endpoint for dSYMs).

We can leave this open to judge the popularity of this request

**Comment by aried3r** [comment]
Hey, Thank you for your reply!

Is this also the reason the ability to upload zip files in the Firebase Console is being phased out? Because it seems that it at least used to be possible to upload zip files directly, even if manually.

**Comment by samedson** [comment]
Yes that's exactly right - we haven't found a suitable way to support that system again, but we are always open to feedback!

**Comment by aried3r** [comment]
Thank you for shedding a little light on the issue! I can only assume what's going on, so I guess only macOS has libraries to convert dSYMs into another format that Firebase can use? Is there any documentation on it?

As for the feedback, it's pretty much this FR, thanks!

**Comment by aried3r** [comment]
I just found this:

> *Note: `upload-symbols`* is in the process of integrating with the rest of the Firebase CLI tools.

https://firebase.google.com/docs/crashlytics/get-deobfuscated-reports#upload-dsyms

Does this mean there will be a CLI tool that can be used outside of having to install the Pod?

**Comment by samedson** [comment]
Yeah at some point we plan to offer it separately as well, like you said. But it's most likely we will only allow uploading iOS symbols on macOS, same as upload-symbols.

**Comment by robmathers** [comment]
Tossing a vote out for this one as well. Would make it much simpler & cheaper to be able to automate it with CI actions on less expensive/more available Linux machines (GitHub Actions, etc.)

**Comment by elenadoty** [comment]
Hi all, 
Just a quick update here - upload symbols still requires libraries that only ship with MacOS, but we're still looking into ways to make it possible to run without these libraries.

**Comment by SzymonGajdzica** [comment]
Any updates on this issue? Waiting for your dsym to be uploaded on your work device is a nightmare. The ability to use CI would be great

**Comment by leontiy** [comment]
A Linux version is still very much needed!

**Comment by mathaeus** [comment]
Any update on that? :)
